llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -903,6 +903,10 @@ Instruction *InstCombinerImpl::foldIntrinsicIsFPClass(IntrinsicInst &II) {
   const FPClassTest OrderedInvertedMask = ~OrderedMask & ~fcNan;
 
   const bool IsStrict = II.isStrictFP();
+  // An fcmp instruction can raise floating-point exceptions, so we can't
+  // introduce new ones if the enclosing function has the strictfp attribute.
+  const bool CanIntroduceFCmp =
+      !II.getFunction()->getAttributes().hasFnAttr(Attribute::StrictFP);
```

Comment:
This should be the same as II.isStrictFP above. However, is.fpclass would never be strict, so I'm not sure why this is checking that.

Reply:
`II.getFunction()` is checking the function which contains this call, not the intrinsic. Since `is.fpclass` can never be strict, I'll replace the old check with the new one, instead of having both.